contracts/contracts/Umbra.sol

```diff
@@ -10,28 +10,55 @@ import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
 contract Umbra is Ownable {
   using SafeMath for uint256;
 
-  struct TokenPayment {
-    address token;
-    uint256 amount;
-  }
+  // =========================================== Events ============================================
 
+  /// @notice Emitted when a payment is sent
   event Announcement(
-    address indexed receiver,
-    uint256 indexed amount,
-    address indexed token,
+    address indexed receiver, // stealth address
+    uint256 amount, // funds
+    address indexed token, // token address or ETH placeholder
     bytes32 pkx, // ephemeral public key x coordinate
-    bytes32 ciphertext
+    bytes32 ciphertext // encrypted entropy and payload extension
+  );
+
+  /// @notice Emitted when a token is withdrawn
+  event TokenWithdrawal(
+    address indexed receiver, // stealth address
+    address indexed acceptor, // destination of funds
```

Comment:
Similar to above, I wonder if there might be some need in the future to filter on the destination where funds were sent. I can't think of an example off the top of my head, but I can imagine the need might exist. The gas overhead for indexing is 375 so the question is whether it's worth it to be able to have the option to do this filtering or not.

Reply:
Now it's 8.2 cents! 😛 

So between these two parameters we're looking at 16 cents per transaction, which sounds small, but (1) it will likely only go up over time, and (2) making everyone pay more for an unknown future use case doesn't seem ideal given the state of gas prices.

Personally I learn towards saving the gas, and if someone needs to filter by stealth address they can just use a third party service to help with parsing the logs, such as Dune

Reply:
Here's a usecase I thought of: if you're writing a contract to act as a withdrawal hook, you might want to filter on all 
withdrawals where your contract was the acceptor.

That said, I completely agree that gas prices are likely to continue going up and we should be super sensitive to increasing gas costs needlessly.

> if someone needs to filter by stealth address or acceptor they can just use a third party service to help with parsing the logs, such as Dune

This would be an argument against ever indexing parameters, right?

I created #119 to track this. And this is another issue that's probably worth mentioning to auditors.